Loop/Managers/TemporaryPresetsManager.swift

```diff
@@ -192,15 +192,29 @@ class TemporaryPresetsManager {
             presets.append(.preMeal(range: preMealTargetRange))
         }
 
-        if let legacyWorkoutTargetRange = settings.workoutTargetRange {
-            let duration = settings.workoutDefaultDuration ?? .indefinite
-            presets.append(.legacyWorkout(
-                range: legacyWorkoutTargetRange,
-                duration: duration.presetDurationType
-            ))
+        presets.append(contentsOf: settings.overridePresets.map { override in
+            if override.id.hasPrefix("activity-"), let activityPreset = ActivityPreset(preset: override) {
+                return .activity(activityPreset)
+            } else {
+                return .custom(override)
+            }
+        })
+        
+        if !settings.overridePresets.contains(where: { $0.id == ActivityPreset.ActivityType.biking.id }) {
+            presets.append(.activity(ActivityPreset(activityType: .biking, preset: ActivityPreset.ActivityType.biking.defaultPreset)))
+        }
+        
+        if !settings.overridePresets.contains(where: { $0.id == ActivityPreset.ActivityType.walking.id }) {
+            presets.append(.activity(ActivityPreset(activityType: .walking, preset: ActivityPreset.ActivityType.walking.defaultPreset)))
+        }
+        
+        if !settings.overridePresets.contains(where: { $0.id == ActivityPreset.ActivityType.jogging.id }) {
+            presets.append(.activity(ActivityPreset(activityType: .jogging, preset: ActivityPreset.ActivityType.jogging.defaultPreset)))
+        }
+        
+        if !settings.overridePresets.contains(where: { $0.id == ActivityPreset.ActivityType.strengthTraining.id }) {
+            presets.append(.activity(ActivityPreset(activityType: .strengthTraining, preset: ActivityPreset.ActivityType.strengthTraining.defaultPreset)))
```

Comment:
Open to ideas on cleaning this up!

Reply:
Can you make it CaseIterable and iterate over the types?